### PR TITLE
Add `index-where` from `racket/list`.

### LINKED ISSUE #1118: index-where missing from racket/list
### What version of Racket are you using?

8.1

### What program did you run?

````
#lang typed/racket/base
(require racket/list)
(index-where '("a" "b") (lambda (x) (and (string? x) (string=? x "b"))))
````

### What should have happened?

Either a type error should have been omitted, or the intended computation should have produced the expected answer `1`, as it does in `#lang racket/base`.

### If you got an error message, please include it here.

````
busted.rkt:3:1: Type Checker: missing type for identifier;
 consider using `require/typed' to import it
  identifier: index-where
  from module: racket/list
  in: index-where
  ...
````

### PR DESCRIPTION
**Pull request by jessealama**
closes #1118

### CODE REVIEW

File: typed-racket-lib/typed-racket/base-env/base-env.rkt [hunk 1]
```diff
@@ -696,6 +696,11 @@
                  (cl->* ((-lst a) Univ . -> . (-opt -Index))
                         ((-lst a) b (-> a b Univ)
                            . -> . (-opt -Index))))]
+[index-where (-poly (a b)
+                    (Univ (-lst a)
```
Comment (samth) [inline typed-racket-lib/typed-racket/base-env/base-env.rkt:700]: What is the first argument of type Univ for?

File: typed-racket-lib/typed-racket/base-env/base-env.rkt [hunk 2]
```diff
@@ -696,6 +696,11 @@
                  (cl->* ((-lst a) Univ . -> . (-opt -Index))
                         ((-lst a) b (-> a b Univ)
                            . -> . (-opt -Index))))]
+[index-where (-poly (a b)
```
Comment (capfredf) [inline typed-racket-lib/typed-racket/base-env/base-env.rkt:699]: looks like we don't need `b` in the binder

File: typed-racket-lib/typed-racket/base-env/base-env.rkt (hunk 1 above)
Comment (jessealama) [reply typed-racket-lib/typed-racket/base-env/base-env.rkt]: oops, not necessary

File: typed-racket-lib/typed-racket/base-env/base-env.rkt (hunk 2 above)
Comment (jessealama) [reply typed-racket-lib/typed-racket/base-env/base-env.rkt]: right, thanks -- dropped

**Comment by samth** [comment]
Thanks

**Pull request merged**